test/Microsoft.SqlTools.ServiceLayer.UnitTests/ObjectExplorer/NodeFilterTests.cs

```diff
@@ -610,5 +610,45 @@ public void TextNumericFilters()
             };
             Assert.Throws<IndexOutOfRangeException>(() => INodeFilter.GetPropertyFilter(filterList, typeof(SqlHistoryTableQuerier), ValidForFlag.All), "Error not thrown when the array contains single value for between operator");
         }
+
+        [Test]
+        public void TestStartsWithFilter()
```

Comment:
Can you add tests for testing the filtering logic itself to show that you're constructing the clauses correctly?